scenedetect/detectors/hash_detector.py

```diff
@@ -0,0 +1,168 @@
+                # Resize image to square to help with DCT
+                imsize = self.hash_size * self.highfreq_factor
+                curr_resized = cv2.resize(curr_gray, (imsize, imsize), interpolation=cv2.INTER_AREA)
+                # Calculate discrete cosine tranformation of the image
+                curr_resized = numpy.float32(curr_resized) / numpy.max(numpy.max(curr_resized))
+                curr_dct = cv2.dct(curr_resized)
+                # Only keep the low frequency information
+                curr_dct_low_freq = curr_dct[:self.hash_size, :self.hash_size]
+                # Calculate the median of the low frequency information
+                curr_med = numpy.median(curr_dct_low_freq)
+                # Transform the low frequency information into a binary image based on > or < median
+                curr_hash = curr_dct_low_freq > curr_med
+                
+                last_hash = self.last_hash
+                
+                if last_hash.size == 0:
+                    # Calculate hash as above
+                    last_gray = cv2.cvtColor(self.last_frame, cv2.COLOR_BGR2GRAY)
+                    last_resized = cv2.resize(last_gray, (imsize, imsize), interpolation=cv2.INTER_AREA)
+                    last_resized = numpy.float32(last_resized) / numpy.max(numpy.max(last_resized))
+                    last_dct = cv2.dct(last_resized)
+                    last_dct_low_freq = last_dct[:self.hash_size, :self.hash_size]
+                    last_med = numpy.median(last_dct_low_freq)
+                    last_hash = last_dct_low_freq > last_med
+                
+                # Hamming distance is calculated to compare to last frame
+                hash_dist = numpy.count_nonzero(curr_hash.flatten() != last_hash.flatten())
+
+                if self.stats_manager is not None:
+                    self.stats_manager.set_metrics(frame_num, {
+                        metric_keys[0]: hash_dist})
+                
+                self.last_hash = curr_hash
+                
+
+            # We consider any frame over the threshold a new scene, but only if
+            # the minimum scene length has been reached (otherwise it is ignored).
+            if hash_dist >= self.threshold and (
+                    (frame_num - self.last_scene_cut) >= self.min_scene_len):
+                cut_list.append(frame_num)
+                self.last_scene_cut = frame_num
+
+            if self.last_frame is not None and self.last_frame is not _unused:
+                del self.last_frame
+            
+        # If we have the next frame computed, don't copy the current frame
+        # into last_frame since we won't use it on the next call anyways.
+        if (self.stats_manager is not None and
+                self.stats_manager.metrics_exist(frame_num+1, metric_keys)):
+            self.last_frame = _unused
+        else:
+            self.last_frame = frame_img.copy()
+
+        return cut_list
```

Comment:
Missing newline at end of file.  Sorry the linter isn't enabled on PRs yet, I'll make a note to fix that to automate this.

Reply:
Done with 0f54efc.